samples/JavaScript/entry.mjs

```diff
@@ -0,0 +1,6 @@
+import bar from './module.mjs';
```

Comment:
can you defined "real", this PR is pre-emptive to make the transition easier. Most projects are using pseudo-ESM with `.js` (and some form of runtime/compile time transform) but `.mjs` will be the same content wise. Though I don't see ESM in the samples either way.

Would using the pseudo-ESM files be reasonable even if they are not to spec?